azurelinuxagent/ga/exthandlers.py

```diff
@@ -929,15 +929,19 @@ def execute_old_handler_command_and_return_if_succeeds(func):
                     func=partial(old_ext_handler_i.disable, extension=old_ext))
 
         ext_handler_i.copy_status_files(old_ext_handler_i)
+        # Invoke the update method of the greater version, both on upgrades and downgrades
         if ext_handler_i.version_gt(old_ext_handler_i):
-            ext_handler_i.update(disable_exit_codes=disable_exit_codes,
-                                 updating_from_version=old_ext_handler_i.ext_handler.version,
-                                 extension=extension)
+            ext_handler_i.update(
+                updating_from_version=old_ext_handler_i.ext_handler.version,
+                updating_to_version=ext_handler_i.ext_handler.version,
+                extension=extension,
+                disable_exit_codes=disable_exit_codes)
         else:
-            updating_from_version = ext_handler_i.ext_handler.version
-            old_ext_handler_i.update(handler_version=updating_from_version,
-                                     disable_exit_codes=disable_exit_codes, updating_from_version=updating_from_version,
-                                     extension=extension)
+            old_ext_handler_i.update(
```

Comment:
is update operation supposed to happen on greater version? or version we update to? The reason I ask we do update on new version for upgrade and old version for downgrade?